Refactor activity panel

furinvader opened the pull request:
General refactoring of the activity panel(s).
Remove code redundancies and implement a directive for it, as previously done for the watchers panel.

furinvader committed:
Refactor watchers panel file structure

furinvader committed:
Implement activity panel directive

furinvader committed:
Extract code to ActivityPanelController

Copy pasta buster

furinvader committed:
Reinitialise activities content after comment update

furinvader committed:
Implement wpActivity service

Move the main logic into this service instead
of having it in the controller.

furinvader committed:
Refactor isInitialActivity method

furinvader committed:
Add order scope attribute

[ci skip]

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://furinvader-refactoring-activity-panel.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://furinvader-refactoring-activity-panel.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

oliverguenther committed:
Refactor aggregation in ActivityService using promises

oliverguenther committed:
Remove embedded activities

furinvader committed:
Add alternative activity panel template for overview panel

furinvader committed:
Fix error in wpActivity service

furinvader committed:
Remove redundant promise implementation

As the resources of the work package links are
promises themselves, there are no other
more promises needed.

furinvader committed:
Update wpActivity.activities without overriding it

furinvader committed:
Prevent reload flash when updating activities

furinvader committed:
Move autocompletePath to activityPanel directive

furinvader committed:
Implement wpActivity info helper object
